src/tests/ftest/pool/create_capacity.py

```diff
@@ -40,24 +310,37 @@ def test_create_pool_quantity(self):
         :avocado: tags=pool
         :avocado: tags=PoolCreateCapacityTests,test_create_pool_quantity
         """
-        # Create some number of pools each using a equal amount of 60% of the
-        # available capacity, e.g. 0.6% for 100 pools.
-        quantity = self.params.get("quantity", "/run/pool/*", 1)
-        storage = self.server_managers[0].get_available_storage()
-        if storage['nvme'] < 750156374016:
+        self.log_step("Retrieving storage usage")
+        storage_usage = self.get_dmg_command().storage_query_usage()
+
+        scm_avail_bytes, nvme_avail_bytes = self.get_available_storage(storage_usage)
+        self.log.debug("Available storage for pools:")
+        self.log.debug("  - SCM:  %s", get_display_size(scm_avail_bytes))
+        self.log.debug("  - NVMe: %s", ", ".join(
+            f"{role}={get_display_size(avail_bytes)}"
+            for role, avail_bytes in nvme_avail_bytes.items()))
+
+        # Evaluate the maximal quantity of pools which can be created
+        if self.md_on_dev == "md_on_scm":
+            quantity_max = self.get_md_on_scm_quantity(storage_usage, nvme_avail_bytes['data'])
+        else:
+            quantity_max = self.get_md_on_ssd_quantity(
+                storage_usage, scm_avail_bytes, nvme_avail_bytes)
+        self.log.debug("At most %d pools can be created", quantity_max)
+        if self.quantity > quantity_max:
             self.log.info(
-                'Reducing pool quantity from %s -> 150 due to insufficient NVMe capacity (%s < '
-                '750156374016)', quantity, storage['nvme'])
-            quantity = 150
+                "Reducing pool quantity from %d -> %d due to insufficient storage capacity",
+                self.quantity, quantity_max)
+            self.quantity = quantity_max
```

Comment:
We should add a check here to make sure the test isn't passing with a pool quantity that doesn't make sense. No more 15%, 20%, or 25% less pools??

Reply:
- [ ] Check minimal quantity of pools

Reply:
> We should add a check here to make sure the test isn't passing with a pool quantity that doesn't make sense. No more 25% less pools specified by `self.quantity`?? How many pools do we end up creating with MD on SDD on a cluster with the smaller 375 drives?

With reducing the RDB size from 1GiB to 128MiB, we should be able to create 200 pools for md-on-scm and also for md-on-ssd.  For md-on-ssd, I am not fully confident as I have done the experiment a long time ago and the calculation of the space needed for each pool is quite complicated.

Reply:
- [ ] Strengthen and simplify the pools creation step

Reply:
> * [x]  Check minimal quantity of pools
> * [x]  Strengthen and simplify the pools creation step

Fixed with commit 4cc33422cf932c174b5edc37683d8ac047013369